library/test/src/lib.rs

```diff
@@ -182,7 +183,7 @@ fn make_owned_test(test: &&TestDescAndFn) -> TestDescAndFn {
 /// Tests is considered a failure. By default, invokes `report()`
 /// and checks for a `0` result.
 pub fn assert_test_result<T: Termination>(result: T) {
-    let code = result.report();
+    let code = result.report().to_i32();
     assert_eq!(
         code, 0,
```

Comment:
pondering: this could use `code, ExitCode::SUCCESS,` instead of `to_i32`, right?  (Might also need `{:?}` instead of `{}` in the message -- but that might be nice anyway as it'd say `ExitCode(10)` and not need to write out "code" and such in prose.)

Reply:
Also, the language here references `non-zero` status codes, but the impression I've got is that `ExitCode::SUCCESS` isn't necessarily a `0`. Do we support any platforms where `SUCCESS` isn't a `0`?

Reply:
`ExitStatus` says we don't for things with `std`, at least: <https://doc.rust-lang.org/std/process/struct.ExitStatus.html#method.success>

Reply:
We cant do this without adding a `PartialEq` impl to `ExitCode`, which is probably justified but I don't want add another API decision to this PR. Is it okay if I do this as a followup?

Reply:
Yup, absolutely ok to think about it later.  Not adding another trait impl is a great reason to not do it here.